packages/services/f-analytics/src/mixins/_tests/analytics.methods.test.js

```diff
@@ -0,0 +1,218 @@
+import Vuex from 'vuex';
+import { shallowMount, createLocalVue } from '@vue/test-utils';
+import analyticsMixin from '../analytics.mixin.vue';
+import {
+    createStore,
+    gtmSettings,
+    $route,
+    $i18n
+} from './helpers/setup';
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+describe('Analytics', () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('methods ::', () => {
+        describe('preparePage ::', () => {
+            let component;
+            let preparePageSpy;
+
+            beforeEach(() => {
+                // Arrange
+                document.head.innerHTML = '';
+                component = {
+                    render () {},
+                    mixins: [analyticsMixin],
+                    store: createStore()
+                };
+
+                component.mixins[0].created = jest.fn(() => true);
+                jest.spyOn(component.mixins[0].methods, 'prepareAnalytics').mockImplementationOnce(() => true);
+                jest.spyOn(component.mixins[0].methods, 'pushAnalytics').mockImplementationOnce(() => true);
+                component.mixins[0].computed.isServerSide = () => false;
+                preparePageSpy = jest.spyOn(component.mixins[0].methods, 'preparePage');
+            });
+
+            it('should append the GTM tag if the dataLayer is not already present', () => {
+                // Arrange
+                window.dataLayer = undefined;
+
+                // Act
+                shallowMount(
+                    component,
+                    {
+                        localVue,
+                        mocks: {
+                            gtmSettings
+                        }
+                    }
+                );
+
+                // Assert
+                expect(preparePageSpy).toHaveBeenCalled();
+                expect(document.head.innerHTML).toContain('src="https://www.googletagmanager.com/gtm.js?id=GTM-123456A"');
```

Comment:
More of a suggestion but would it be cleaner to just check that the method returns true or falsey if `window.dataLayer` isn't defined? That way it can save the setup of checking for the structure of the innerHTML code. (e.g the GTM id could change and this test would fail)

Reply:
I want to make sure the GTM markup is not changed (broken) by anyone so that's why I check for specifics. The GTM id is mocked.